lib/assertions.js

```diff
@@ -1192,7 +1198,9 @@ module.exports = expect => {
         err => {
           expect.fail(output => {
             output.append(
-              expect.standardErrorMessage(output.clone(), { compact: true })
+              expect.standardErrorMessage(output.clone(), {
+                compact: err.hasDiff()
```

Comment:
Might it be worth keeping the `hasDiff()` conditional consistent with the others added in this PR?

Reply:
Right, I guess a non-Unexpected error could slip through and be caught here. Good shout.

Reply:
57da7c49